packages/playwright/src/fixture.ts

```diff
@@ -0,0 +1,296 @@
+  event: "error";
+  test: TestIdData;
+}
+
+export type FixtureEvent = FixtureStepStartEvent | FixtureStepEndEvent | ReporterErrorEvent;
+
+const debug = dbg("replay:playwright:fixture");
+
+function ReplayAddAnnotation([event, id, data]: any) {
+  // @ts-ignore
+  window.__RECORD_REPLAY_ANNOTATION_HOOK__?.("replay-playwright", {
+    event,
+    id,
+    data: data ? JSON.parse(data) : undefined,
+  });
+}
+
+function getLastStepId(testInfo: any) {
+  return testInfo._steps[testInfo._steps.length - 1].stepId;
+}
+
+function isReplayAnnotation(params?: any) {
+  return params?.expression?.includes("ReplayAddAnnotation");
+}
+
+function parseLocation(stack?: string) {
+  const pattern = /\/([^\/]+):(\d+):(\d+)$/;
+  const firstLine = stack?.split("\n").find(l => pattern.test(l));
+  const match = firstLine?.match(pattern);
+
+  if (!match) {
+    return {
+      line: undefined,
+      column: undefined,
+    };
+  }
+
+  return {
+    line: parseInt(match[2]),
+    column: parseInt(match[3]),
+  };
+}
+
+function parseError<T extends { name?: string; message: string; stack?: string }>(error: T) {
+  if (!error) {
+    return;
+  }
+
+  const location = parseLocation(error.stack);
+
+  return {
+    name: error.name || "Error",
+    message: error.message,
+    line: location?.line,
+    column: location?.column,
+  };
+}
+
+function maybeMonkeyPatchTestInfo(
+  testInfo: TestInfoInternal,
+  addStepHandler: (step: TestInfoStep) => void,
+  stepEndHandler: TestInfoInternal["_onStepEnd"]
+) {
+  if (testInfo._addStep) {
+    const original = testInfo._addStep.bind(testInfo);
+    testInfo._addStep = (data, parentStep) => {
+      const result = original(data, parentStep);
+      addStepHandler?.(result);
+
+      return result;
+    };
+  }
+
+  if (testInfo._onStepEnd) {
+    const original = testInfo._onStepEnd.bind(testInfo);
+    testInfo._onStepEnd = step => {
+      const result = original(step);
+      stepEndHandler?.(step);
+
+      return result;
+    };
+  }
+}
+
+export async function replayFixture(
+  { playwright, page }: { playwright: any; page: Page },
+  use: () => Promise<void>,
+  testInfo: TestInfo
+) {
+  debug("Setting up replay fixture");
+
+  const port = getServerPort();
+  const ws = new WebSocket(`ws://localhost:${port}`);
+  const expectSteps = new Set<string>();
+  let currentStepId: string | undefined;
+
+  await new Promise<void>((resolve, reject) => {
+    ws.on("open", () => resolve());
+    ws.on("error", () => reject("Socket errored"));
+  });
+
+  const testIdData: TestIdData = {
+    id: 0,
+    attempt: testInfo.retry + 1,
+    source: {
+      title: testInfo.title,
+      scope: testInfo.titlePath.slice(3, -1),
+    },
+  };
+
+  function addAnnotation(event: string, id?: string, data?: Record<string, any>) {
+    if (id) {
+      page
+        .evaluate(ReplayAddAnnotation, [event, id, JSON.stringify({ ...data, test: testIdData })])
+        .catch(e => console.error);
```

Comment:
Shouldn't this be `catch(e => console.error(e))` or `catch(console.error)`?

Reply:
Yup!